doc/constraint_causal_discovery.rst

```diff
@@ -0,0 +1,136 @@
+.. _constraint_causal_discovery:
+
+==================================
+Constraint-based causal discovery
+==================================
+.. automodule:: dodiscover.constraint
+    :no-members:
+    :no-inherited-members:
+.. currentmodule:: dodiscover
+
+The following are a set of methods intended for (non-parametric) structure learning
+of causal graphs (i.e. causal discovery) given observational and/or interventional data
+by checking constraints in the form of conditional independences (CI). At a high level
+all constraint-based causal discovery, tests CI statements, where we use different CI
```

Comment:
The "At a high level.." sentence is difficult to read. What about: "At a high level, all constraint-based causal discovery algorithms test CI statements, where we use..."

Reply:
Done.